src/main/java/DateParser.java

```diff
@@ -0,0 +1,96 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DateParser {
+
+    private static SimpleDateFormat formatter1 = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
```

Comment:
I like how you took flexibility of the date inputs into account.
However, I do think you shouldn't be using this method to enhance flexibility. Unfortunately, I face the same problem and do not have a viable solution. Perhaps you could try to seek help from your tutors or the forum for a more viable solution?